### PR TITLE
acl: remove legacy ACL.Apply RPC

### PR DESCRIPTION
**Pull request by dnephin**
Branched from #10986

This PR removes the remaining tests for `ACL.Apply` (and updates one for ACLResolver to not use ACL.Apply), and then replaces the implementation with an error.

### CODE REVIEW

File: agent/setup.go [hunk 1]
```diff
@@ -296,7 +296,6 @@ func getPrometheusDefs(cfg lib.TelemetryConfig) ([]prometheus.GaugeDefinition, [
 		HTTPSummaries,
 		consul.ACLSummaries,
 		consul.ACLEndpointSummaries,
-		consul.ACLEndpointLegacySummaries,
```
Comment (rboyer) [inline agent/setup.go:299]: If you hadn't already, this should go on the short list of stuff that needs cleaned up on the website page for metrics.

**Review by rboyer — APPROVED** [review]
LGTM

File: agent/setup.go (hunk 1 above)
Comment (dnephin) [reply agent/setup.go:299]: Oh, good call. I can probably add that to this PR

**Review by karl-cardenas-coding — APPROVED** [review]
Good from a doc perspective

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/460414.

**Pull request merged**